docs/pkg/client/events.md

```diff
@@ -0,0 +1,421 @@
+
+```mermaid
+---
+title: Component Diagram Legend
+---
+
+flowchart
+
+  a[Component A]
+  b[Component B]
+  c[Component C]
+  d[Component D]
+
+  a --"A uses B via B#MethodName()"--> b
+  a =="A returns C from A#MethodName()"==> c
+  b -."B uses D via network IO".-> d
+```
+
+#### Events Query Client
+
+```mermaid
+---
+title: EventsQueryClient Components
+---
+
+flowchart
+
+  subgraph comet[Cometbft Node]
+    subgraph rpc[JSON-RPC]
+      sub[subscribe endpoint]
+    end
+  end
+
+  subgraph eqc[EventsQueryClient]
+    q1_eb[EventsBytesObservable]
+    q1_conn[Connection]
+    q1_dial[Dialer]
+  end
+
+  q1_obsvr1[Observer 1]
+  q1_obsvr2[Observer 2]
+
+
+  q1_obsvr1 --"#Subscribe()"--> q1_eb
+  q1_obsvr2 --"#Subscribe()"--> q1_eb
+
+
+  q1_dial =="#DialContext()"==> q1_conn
+  q1_eb --"#Receive()"--> q1_conn
+
+  q1_conn -.-> sub
+```
+
+#### Events Replay Client
+
+```mermaid
+---
+title: EventsReplayClient Components
+---
+
+flowchart
+    slice_cons[Events Slice Consumer]
+
+    subgraph obs_cons[Events Observable Consumer]
+        evt_obs[Events Observer]
+    end
+
+    subgraph erc[EventsReplayClient]
+        cache[Events Replay Observable Cache]
+    end
+
+    chain[CometBFT Node]
+
+    evt_re_obs[Events Replay Observable]
+
+    eqc[Events Query Client]
+
+
+    slice_cons --"#LastNEvents()"--> erc
+    obs_cons --"#EventsSequence()"--> erc
+    erc =="#EventsSequence()"==> evt_re_obs
+    evt_obs --"#Subscribe()"--> evt_re_obs
+
+    erc --"#EventBytes()"--> eqc
+
+    eqc -.-> chain
+```
+
+### Subscriptions
+
+_TODO_DOCUMENTATION(@bryanchriswhite): Add Legend_
+
+```mermaid
+---
+title: Event Subscription Data Flow
+---
+
+flowchart
+
+  subgraph comet[Cometbft Node]
+      subgraph rpc[JSON-RPC]
+          sub[subscribe endpoint]
+      end
+  end
+
+  subgraph eqc[EventsQueryClient]
+      subgraph q1[Query 1]
+          q1_eb[EventsBytesObservable]
+          q1_conn[Connection]
+      end
+      subgraph q2[Query 2]
+          q2_conn[Connection]
+          q2_eb[EventsBytesObservable]
+      end
+  end
+
+  q1_obsvr1[Query 1 Observer 1]
+  q1_obsvr2[Query 1 Observer 2]
+  q2_obsvr[Query 2 Observer]
+
+  q1_eb -.-> q1_obsvr1
+  q1_eb -.-> q1_obsvr2
+  q2_eb -.-> q2_obsvr
+
+
+  q1_conn -.-> q1_eb
+  q2_conn -.-> q2_eb
+
+  sub -.-> q1_conn
+  sub -.-> q2_conn
+```
+
+### Events Decoding
+
+_TODO_DOCUMENTATION(@bryanchriswhite,@h5law): Add Legend_
+
+_TODO_DOCUMENTATION(@bryanchriswhite,@h5law): separate into separate diagrams that detail each
+step of the process (reconnection, decoding, publishing, etc.)_
+
+```mermaid
+---
+title: Event Bytes To Type Event Sequence
+---
+
+sequenceDiagram
+    participant Q as EventsQueryClient
+    participant D as EventDecoder
+    Q ->>+ D: EventBytes
+    D ->> D: DecoderFn()
+    D ->>- Q: resubscribeOnError()
+    participant P as Replay Observable Cache Publish Channel
+    participant O as Replay Observable Cache
+    D ->> P: Typed Events
+    P ->> O: Typed Events
+```
```

Comment:
Let's shelve this diagram for now (i.e. remove and iterate in [notion](https://www.notion.so/Clients-a24e75fe6f70450890c8e12092d5f8e1?d=653b16f968c34e09bed1fe06cb0e0439&pvs=4#9c938bc617374de0a95d911e1d6d2641)), it's not ready and isn't worth blocking this PR over.

Reply:
:pray: